examples/src/bin/encodebin.rs

```diff
@@ -0,0 +1,271 @@
+#[macro_use]
+extern crate gstreamer as gst;
+use gst::prelude::*;
+
+extern crate gstreamer_pbutils as pbutils;
+use pbutils::EncodingProfileBuilder;
+
+extern crate glib;
+
+use std::env;
+use std::error::Error as StdError;
+#[cfg(feature = "v1_10")]
+use std::sync::{Arc, Mutex};
+
+extern crate failure;
+use failure::Error;
+
+#[macro_use]
+extern crate failure_derive;
+
+#[path = "../examples-common.rs"]
+mod examples_common;
+
+#[derive(Debug, Fail)]
+#[fail(display = "Missing element {}", _0)]
+struct MissingElement(&'static str);
+
+#[derive(Debug, Fail)]
+#[fail(display = "Received error from {}: {} (debug: {:?})", src, error, debug)]
+struct ErrorMessage {
+    src: String,
+    error: String,
+    debug: Option<String>,
+    #[cause]
+    cause: glib::Error,
+}
+
+fn configure_encodebin(encodebin: &gst::Element) -> Result<(), Error> {
+    let audio_profile = pbutils::EncodingAudioProfileBuilder::new()
+        .format(&gst::Caps::new_simple(
+            "audio/x-vorbis",
+            &[],
+        ))
+        .presence(0)
+        .build()?;
+
+    let video_profile = pbutils::EncodingVideoProfileBuilder::new()
+        .format(&gst::Caps::new_simple(
+            "video/x-theora",
+            &[],
+        ))
+        .presence(0)
+        .build()?;
+
+    let container_profile = pbutils::EncodingContainerProfileBuilder::new()
+        .name("container")
+        .format(&gst::Caps::new_simple(
+            "video/x-matroska",
+            &[],
+        ))
+        .add_profile(&(video_profile.upcast()))
+        .add_profile(&(audio_profile.upcast()))
+        .build()?;
+
+    encodebin.set_property("profile", &container_profile)?;
```

Comment:
This should be an `expect` or `unwrap`. It only can possibly fail here if you did a programming mistake